llvm/lib/Transforms/IPO/LowerTypeTests.cpp

```diff
@@ -1837,6 +1851,16 @@ LowerTypeTestsModule::LowerTypeTestsModule(
   }
   OS = TargetTriple.getOS();
   ObjectFormat = TargetTriple.getObjectFormat();
+
+  // Function annotation describes or applies to function itself, and
+  // shouldn't be associated with jump table thunk generated for CFI.
+  GlobalAnnotation = M.getGlobalVariable("llvm.global.annotations");
+  auto *C = dyn_cast_or_null<Constant>(GlobalAnnotation);
```

Comment:
This cast makes no sense. GlobalAnnotation is a GlobalVariable, which is a subclass of Constant.

Reply:
This is copied from `llvm/lib/Transform/IPO/Annotation2Metadata.cpp`, at line 33.